Project-Hotpot/Project-Hotpot/APIManager.swift

```diff
@@ -0,0 +1,209 @@
+//
+//  APIManager.swift
+//  Project-Hotpot
+//
+//  Created by Harini Sundaram on 7/11/22.
+//
+
+import Foundation
+import UIKit
+
+class APIManager: NSObject {
+//    @IBOutlet weak var songTitle: UILabel!
+    var curr_song_label:String?
+    
+//    class NetworkManager {
+
+        // MARK: - Properties
+
+        private static var sharedAPIManager: APIManager = {
+            let SpotifyManager = APIManager()
+            return SpotifyManager
+        }()
+
+        // MARK: - Initializers
+        //doing this way so that only 1 instance can be created
+        override private init() {
+            NSLog("API Manager Initialized")
+        }
+
+        // MARK: - Accessors
+        
+        class func shared() -> APIManager {
+            return sharedAPIManager
+        }
+
+    
+    var lastPlayerState: SPTAppRemotePlayerState?
+    // MARK: - Spotify Authorization & Configuration
+    var responseCode: String? {
+        didSet {
+            fetchAccessToken { (dictionary, error) in
+                if let error = error {
+                    print("Fetching token request error \(error)")
```

Comment:
Better to consistently use NSLog?

Reply:
Yes, I'd say stick with one. Here's an interesting article comparing the two: https://stevenpcurtis.medium.com/logging-in-swift-d9b59146ff00

Either one is perfectly fine in my opinion, just a matter of how much additional detail you like in your logging method.